#! /bin/sh -e
## 10makefiles.dpatch by Petter Reinholdtsen <pere@hungry.com>
##
## All lines beginning with `## DP:' are a description of the patch.
##
## DP: Build/install rules for sql-ledger

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
       -patch)   ( patch $patch_opts -p1 ) < $0 ;;
       -unpatch) ( patch $patch_opts -p1 -R ) < $0 ;;
        *)
                echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
                exit 1;;
esac

exit 0
@DPATCH@
--- sql-ledger-2.2.6.orig/bin/Makefile
+++ sql-ledger-2.2.6/bin/Makefile
@@ -0,0 +1,18 @@
+PACKAGE = sql-ledger
+
+# Assumption: only the menu/login/admin.pl differs between each directory
+MODULES = $(shell find mozilla/ -name '*.pl' | sed -e 's|mozilla/||' | grep -v menu.pl | grep -v login.pl | grep -v admin.pl) 
+SUBDIRS = lynx mozilla
+
+all:
+
+clean:
+
+install: 
+	mkdir -p $(DESTDIR)/usr/share/sql-ledger/bin
+	cp -r $(SUBDIRS) $(DESTDIR)/usr/share/sql-ledger/bin
+
+	cd $(DESTDIR)/usr/share/sql-ledger ; \
+	for module in $(MODULES) ; do \
+	  (cd bin/lynx ; ln -f ../mozilla/$$module) ;\
+	done ;
--- sql-ledger-2.2.6.orig/doc/Makefile
+++ sql-ledger-2.2.6/doc/Makefile
@@ -0,0 +1,13 @@
+PACKAGE = sql-ledger
+
+FILES	= $(shell ls * | grep -v Makefile | grep -v COPYING | grep -v COPYRIGHT)
+
+all:
+
+clean:
+
+install: 
+	mkdir -p $(DESTDIR)/usr/share/doc/sql-ledger
+	cp -a $(FILES) $(DESTDIR)/usr/share/doc/sql-ledger
+	
+#for ALL in locale/*/COPYING ; do HEAD=$(grep -n "# This program is free software" $ALL |cut -f1 -d:); let HEAD=$HEAD-1; head -$HEAD $ALL | tail +4 ; done | grep -v "^#$"
--- sql-ledger-2.2.6.orig/sql/Makefile
+++ sql-ledger-2.2.6/sql/Makefile
@@ -0,0 +1,11 @@
+PACKAGE = sql-ledger
+
+FILES = *.sql
+
+all:
+
+clean:
+
+install: 
+	mkdir -p $(DESTDIR)/usr/share/sql-ledger/sql
+	cp $(FILES) $(DESTDIR)/usr/share/sql-ledger/sql
--- sql-ledger-2.2.6.orig/templates/Makefile
+++ sql-ledger-2.2.6/templates/Makefile
@@ -0,0 +1,14 @@
+PACKAGE = sql-ledger
+
+FILES	= *.html *.tex *.txt *.eps *.png
+	  
+all:
+
+clean:
+
+install: 
+	mkdir -p $(DESTDIR)/var/lib/sql-ledger/templates
+	cp -a $(FILES) $(DESTDIR)/var/lib/sql-ledger/templates
+	chown www-data:www-data $(DESTDIR)/var/lib/sql-ledger/templates
+	ln -sf /var/lib/sql-ledger/templates $(DESTDIR)/usr/share/sql-ledger/
+	
--- sql-ledger-2.2.6.orig/css/Makefile
+++ sql-ledger-2.2.6/css/Makefile
@@ -0,0 +1,13 @@
+PACKAGE = sql-ledger
+
+FILES	= $(shell ls sql-ledger*.css)
+
+all:
+
+clean:
+
+install: 
+	mkdir -p $(DESTDIR)/var/lib/sql-ledger/css
+	cp $(FILES) $(DESTDIR)/var/lib/sql-ledger/css
+	chown -R www-data:www-data $(DESTDIR)/var/lib/sql-ledger/css
+	ln -sf /var/lib/sql-ledger/css $(DESTDIR)/usr/share/sql-ledger/
--- sql-ledger-2.2.6.orig/users/Makefile
+++ sql-ledger-2.2.6/users/Makefile
@@ -0,0 +1,13 @@
+PACKAGE = sql-ledger
+
+FILES	= sql-ledger.eps  sql-ledger.png
+
+all:
+
+clean:
+
+install: 
+	mkdir -p $(DESTDIR)/var/lib/sql-ledger/users
+	cp -r $(FILES) $(DESTDIR)/var/lib/sql-ledger/users/
+	chown www-data:www-data $(DESTDIR)/var/lib/sql-ledger/users
+	ln -sf /var/lib/sql-ledger/users $(DESTDIR)/usr/share/sql-ledger/
--- sql-ledger-2.2.6.orig/SL/Makefile
+++ sql-ledger-2.2.6/SL/Makefile
@@ -0,0 +1,11 @@
+PACKAGE = sql-ledger
+
+FILES  := $(shell ls *.pm)
+
+all:
+
+clean:
+
+install:
+	mkdir -p $(DESTDIR)/usr/share/sql-ledger/SL
+	cp -r $(FILES) $(DESTDIR)/usr/share/sql-ledger/SL
--- sql-ledger-2.2.6.orig/locale/Makefile
+++ sql-ledger-2.2.6/locale/Makefile
@@ -0,0 +1,43 @@
+PACKAGE = sql-ledger
+
+LOCALES	= $(shell ls */LANGUAGE | cut -d/ -f1)
+
+localedir  = /usr/share/sql-ledger/locale
+dlocaledir = $(DESTDIR)$(localedir)
+srctopdir := $(shell pwd)/..
+
+all:
+
+clean:
+
+install: 
+	mkdir -p $(dlocaledir)
+	set -e ; \
+	for lang in $(LOCALES) ; do \
+	 mkdir -p $(dlocaledir)/$$lang ; \
+	 for file in `find $$lang/. -type f | grep -v COPYING`; do \
+	   if [ -f $$file ] ; then \
+	     cp $$file $(dlocaledir)/$$lang/. ; \
+	   fi ; \
+	 done ; \
+	 if [ -h $$lang/Num2text ] ; then  \
+	   ln -sf ../es/Num2text $(dlocaledir)/$$lang/Num2text ; \
+	 elif [ -h $$lang/Num2text ] ; then  \
+	   cp $$lang/Num2text $(dlocaledir)/$$lang/. ; \
+	 fi ; \
+	done
+	
+	cp de/locales.pl $(dlocaledir)/de
+	set -e ; for lang in $(LOCALES) ; do \
+	  if [ -d $(dlocaledir)/$$lang -a \
+	       -f $(dlocaledir)/$$lang/LANGUAGE -a \
+	       ! -f $(dlocaledir)/$$lang/locales.pl ] ; then \
+	    ln -sf ../de/locales.pl $(dlocaledir)/$$lang/locales.pl ;\
+	fi ; \
+	done
+	
+	set -e ; for lang in $(LOCALES) ; do \
+	  (cd $(dlocaledir)/$$lang && \
+	   perl -I $(srctopdir) ./locales.pl || exit 1) ; \
+	done
+
--- sql-ledger-2.2.6.orig/Makefile
+++ sql-ledger-2.2.6/Makefile
@@ -0,0 +1,36 @@
+PACKAGE = sql-ledger
+
+# Assumption: all .pl except am/login/admin/setup are modules
+MODULES = $(shell ls *.pl | grep -v -E '(am|login|admin|setup)\.pl')
+SUBDIRS = bin doc SL locale sql templates users css
+
+LIB_FILES = am.pl favicon.ico login.pl menu.ini setup.pl \
+            sql-ledger.conf.default VERSION index.html \
+            images
+
+
+CFG_FILES = sql-ledger-httpd.conf sql-ledger.conf
+
+all:
+
+install: 
+	mkdir -p $(DESTDIR)/usr/share/sql-ledger
+	cp -a $(LIB_FILES) $(DESTDIR)/usr/share/sql-ledger/
+	
+	
+
+	cd $(DESTDIR)/usr/share/sql-ledger ; \
+	ln -f login.pl admin.pl ; \
+	set -e ; for module in $(MODULES) ; do \
+	  ln am.pl $$module ; \
+	done ; \
+	ln -sf /etc/sql-ledger/sql-ledger.conf 
+
+	set -e ; for subdir in $(SUBDIRS) ; do \
+	  $(MAKE) -C $$subdir $@ ; \
+	done
+
+	mkdir -p $(DESTDIR)/etc/sql-ledger
+	cp $(CFG_FILES) $(DESTDIR)/etc/sql-ledger
+
+clean:
--- sql-ledger-2.2.6.orig/sql-ledger-httpd.conf
+++ sql-ledger-2.2.6/sql-ledger-httpd.conf
@@ -0,0 +1,8 @@
+
+
+Alias /sql-ledger/ /usr/share/sql-ledger/
+
+<Directory /usr/share/sql-ledger>
+  AddHandler cgi-script .pl
+  Options ExecCGI Includes FollowSymlinks
+</Directory>
